src-electron/rest/user-data.js

```diff
@@ -1008,6 +1119,56 @@ function httpPostDuplicateEndpointType(db) {
   }
 }
 
+/**
+ * Update feature map attribute with given new value
+ *
+ * @param {*} db
+ * @returns status of the update
+ */
+function httpPatchUpdateBitOfFeatureMapAttribute(db) {
+  return async (request, response) => {
+    let { featureMapAttributeId, newValue } = request.body
+    let updated = await queryConfig.updateEndpointTypeAttribute(
+      db,
+      featureMapAttributeId,
+      [['defaultValue', newValue]]
+    )
+    response.status(StatusCodes.OK).json({
+      succesesful: updated > 0
+    })
+  }
+}
+
+/**
+ * Check if conformance data exists in the database
+ *
+ * @param {*} db
+ * @returns boolean value of data exist or not
+ */
+function httpGetConformDataExists(db) {
+  return async (request, response) => {
+    let conformDataExists = await queryFeature.checkIfConformanceDataExist(db)
+    response.status(StatusCodes.OK).json(conformDataExists)
+  }
+}
+
+/**
+ * Set warning for the required element, and delete its existing warning if any.
+ *
+ * @param {*} db
+ * @returns response of setting the warning notification
+ */
+function httpPostRequiredElementWarning(db) {
+  return async (request, response) => {
+    let resp = await querySessionNotification.setRequiredElementWarning(
+      db,
+      request.body,
```

Comment:
It seems unsafe and suspicious to post the whole request body straight into the query. This feels wrong somehow.

Reply:
I improved this by only getting relevant variables passed from frontend, creating a new object out of them, and passing the new object to backend